### PR TITLE
rust: add gnullvm target for CLANG{32,64} envs

### PR DESCRIPTION
**Pull request by mati865**
Prerequisite to disable bootstrap for CLANG envs: #219002

### CODE REVIEW

**Comment by ognevny** [comment]
maybe some config should be changed via config.toml

File: mingw-w64-rust/PKGBUILD [hunk 1]
```diff
@@ -213,6 +213,11 @@ build() {
     sed -i '/target = \[/a\  "wasm32-unknown-unknown", "wasm32-wasip1", "wasm32-wasip1-threads", "wasm32-wasip2",' "${srcdir}/${_realname}c-${pkgver}-src/config.toml"
   fi
 
+  # Add target *-gnullvm
+  if [[ $MINGW_PACKAGE_PREFIX == *-clang-i686 || $MINGW_PACKAGE_PREFIX == *-clang-x86_64 ]]; then
+    sed -i '/target = \[/a\  "${CARCH}-pc-windows-gnullvm",' "${srcdir}/${_realname}c-${pkgver}-src/config.toml"
```
Comment (ognevny) [inline mingw-w64-rust/PKGBUILD:218]:
```suggestion
    sed -i "/target = \[/a\  \"${CARCH}-pc-windows-gnullvm\"," "${srcdir}/${_realname}c-${pkgver}-src/config.toml"
```

**Comment by ognevny** [comment]
it now builds, can be squahed & merged?

**Comment by mati865** [comment]
> it now builds, can be squashed & merged?

Thanks, I had tested only x86_64 build and tried to quickly address CI errors.
I'll try the 32-bit build first and then we can move on.

> maybe some config for gnullvm should be changed via config.toml

This is PR is only for adding additional core and std libs that will be removed in the next PR. If we want additional changes, they should happen in subsequent PRs.

**Comment by ognevny** [comment]
we can move forward with this IMO. anyway clang32 rust ill be dropped at all

**Comment by mati865** [comment]
Agreed

**Comment by lazka** [comment]
Could this error be related? https://github.com/msys2/MINGW-packages/actions/runs/11334394015/job/31520493787?pr=21684

```
  librsvg-2.59.1/meson.build:345:4: ERROR: Problem encountered: error[E0463]: can't find crate for `std`
    |
    = note: the `x86_64-pc-windows-gnu` target may not be installed
    = help: consider downloading the target with `rustup target add x86_64-pc-windows-gnu`
  
  error: aborting due to 1 previous error
  
  For more information about this error, try `rustc --explain E0463`.
  ```

**Comment by mati865** [comment]
Yes, something is passing target that is now wrong: `Running command: D:/M/msys64/clang64/bin/python.exe C:/_/B/src/librsvg-2.59.1/meson/query-rustc.py D:\M\msys64\clang64\bin/rustc.EXE --target x86_64-pc-windows-gnu --query=native-static-libs`

**Comment by lazka** [comment]
Ah, I see, thanks!

**Pull request merged**